### PR TITLE
Reference tidying

### PR DESCRIPTION
**Pull request by graydon**
This just deletes some egregious lies and obsolete terminology -- all of which I originally wrote -- from the reference. I expect the reference itself will be deleted soon enough, but I found myself gritting teeth over these bits too much to let them into a 1.0 release.

### CODE REVIEW

**Comment by steveklabnik** [comment]
❤️❤️❤️❤️❤️❤️❤️❤️👌👍

File: src/doc/reference.md [hunk 1]
```diff
@@ -3888,7 +3883,7 @@ Box<i32>, y: Box<i32>)` declare one mutable variable `x` and one immutable
 variable `y`).
 
 Methods that take either `self` or `Box<Self>` can optionally place them in a
-mutable slot by prefixing them with `mut` (similar to regular arguments):
+mutable variable by prefixing them with `mut` (similar to regular arguments):
```
Comment (bombless) [inline src/doc/reference.md]: Actually I think the original "slot" might be more accurate.
If I have to change "slot" to something I would use "binding" in this case.
See https://github.com/rust-lang/rust/pull/24149

Comment (graydon) [inline src/doc/reference.md]: Sure, binding is fine. Slot is just very old terminology, the only uses in this sense remaining are weird old corners of trans inside the compiler. In the docs it's a dead term, sticks out.

**Pull request merged**